packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,511 @@
+ * @property {string} IP_V6 - Represents the IPv6 address family.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-addressfamily
+ */
+export enum AddressFamily {
+  /**
+   * Represents the IPv4 address family.
+   * Allowed under public and private pool.
+   */
+  IP_V4 = 'ipv4',
+
+  /**
+   * Represents the IPv6 address family.
+   * Only allowed under public pool.
+   */
+  IP_V6 = 'ipv6',
+}
+
+/**
+ * The IP address source for pools in the public scope.
+ * Only used for provisioning IP address CIDRs to pools in the public scope.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-publicipsource
+ */
+export enum IpamPoolPublicIpSource {
+  /**
+   * BYOIP Ipv6 to be registered under IPAM
+   */
+  BYOIP = 'byoip',
+
+  /**
+   * Amazon Provided Ipv6 range
+   */
+  AMAZON = 'amazon',
+}
+
+/**
+ * Limits which service in AWS that the pool can be used in
+ */
+export enum AwsServiceName {
+  /**
+   *  Allows users to use space for Elastic IP addresses and VPCs
+   */
+  EC2 = 'ec2',
+}
+
+/**
+ * Options to create a new Ipam in the account
+ */
+export interface IpamProps{
+
+  /**
+   * The operating Regions for an IPAM.
+   * Operating Regions are AWS Regions where the IPAM is allowed to manage IP address CIDRs
+   * For more information about operating Regions, see [Create an IPAM](https://docs.aws.amazon.com//vpc/latest/ipam/create-ipam.html) in the *Amazon VPC IPAM User Guide* .
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipam.html#cfn-ec2-ipam-operatingregions
+   * @default Stack.region if defined else []
+   */
+  readonly operatingRegion?: string[];
+
+  /**
+   * Name of IPAM that can be used for tagging resource
+   * @default none
+   */
+  readonly ipamName?: string;
+}
+
+/**
+ * Options for configuring an IPAM pool.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ *
+ * @interface PoolOptions
+ * @property {AddressFamily} addressFamily - The address family of the pool (ipv4 or ipv6).
+ * @property {CfnIPAMPool.ProvisionedCidrProperty[]} [provisionedCidrs] - Information about the CIDRs provisioned to the pool.
+ * @property {string} [locale] - The locale (AWS Region) of the pool.
+ * @property {IpamPoolPublicIpSource} [publicIpSource] - The IP address source for pools in the public scope.
+ * @property {string} [awsService] - Limits which AWS service can use the pool.
+ */
+export interface PoolOptions{
+
+  /**
+   * addressFamily - The address family of the pool (ipv4 or ipv6).
+   */
+  readonly addressFamily: AddressFamily;
+
+  /**
+   * [provisionedCidrs] - Information about the CIDRs provisioned to the pool.
+   * @default - No CIDRs are provisioned
+   * @see CfnIPAMPool.ProvisionedCidrProperty
+   */
+  readonly provisionedCidrs?: string[];
```

Comment:
why this should be passed by customer, I think customer will use provisionCidr function to do this, am I correct?

Reply:
this can be given for ipv4 by the customer currently,     
const pool = ipam.privateScope.addPool('Private', {
      addressFamily: vpc.AddressFamily.IP_V4,
      provisionedCidrs: ['10.2.0.0/16'],
      locale: 'us-west-2',
    });

Reply:
it can accept Ipv4 and ipv6, but since we don't support byoip for ipv6 at this point, renamed it to ipv4provisioned cidr and will add a new field in future for ipv6. For validation, will cover that in IPAM class a whole.